### PR TITLE
2016r4 updates

### PR DESCRIPTION
**Pull request by natke**
Updated this library with recent changes to the core ItemSense software, and fixed bugs, including:
- new JobStatus values
- new JobStopReason field
- corrected incorrect URLs
- added sub-configuration element to ReaderConfiguration
- ensured that unknown enums will not cause deserialization errors
- deprecated unknown values and invalid methods
- added floor to item data

Added many new unit tests

### CODE REVIEW

File: src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java [hunk 1]
```diff
@@ -58,7 +71,34 @@ public Response get(Map<String, Object> queryParams, String path, WebTarget targ
...
     }
 
+    public T readObjectFromString(String string) {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationFeature.READ_UNKNOWN_ENUM_VALUES_AS_NULL, true);
+
+        try {
+            return mapper.readValue(string, type);
+        } catch (IOException ioe ) {
+            System.err.println("Could not read class from string");
```
Comment (bholz) [inline src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java]: Does the library not use a logger? Might want to consider using one if we don't.

File: src/test/java/com.impinj.itemsense.client/coordinator/ReaderConfigurationControllerTest.java [hunk 2]
```diff
@@ -72,26 +73,45 @@ public void GetReaderConfigurationsTest(){
         Assert.assertEquals(configurations.size(), 1);
         Assert.assertThat(configurations, instanceOf(ArrayList.class));
         Assert.assertThat(configurations.get(0), instanceOf(ReaderConfiguration.class));
-        Assert.assertEquals(configurations.get(0),testReaderConfiguration);
+        //Assert.assertEquals(testReaderConfiguration, configurations.get(0));
```
Comment (bholz) [inline src/test/java/com.impinj.itemsense.client/coordinator/ReaderConfigurationControllerTest.java]: can this line be removed?

File: src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java (hunk 1 above)
Comment (natke) [inline src/main/java/com/impinj/itemsense/client/helpers/RestApiHelper.java]: Good spot. I've fixed this.

File: src/test/java/com.impinj.itemsense.client/coordinator/ReaderConfigurationControllerTest.java (hunk 2 above)
Comment (natke) [inline src/test/java/com.impinj.itemsense.client/coordinator/ReaderConfigurationControllerTest.java]: Yes. Fixed

**Pull request merged**